### PR TITLE
Adjust completion for explicit interface implementations

### PR DESCRIPTION
**Pull request by AlekseyTs**
- Enable type recommendation within interfaces.
- Do not recommend non-implementable members.

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProviderTests.cs [hunk 1]
```diff
@@ -157,5 +220,189 @@ class Bar : I1
...
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            await VerifyItemExistsAsync(markup, "Goo1()"); // PROTOTYPE(DefaultInterfaceImplementation): Should use VerifyItemIsAbsentAsync instead after test infrastructure is changed to actually build two distinct projects from the XML 
+            await VerifyItemExistsAsync(markup, "Prop1"); // PROTOTYPE(DefaultInterfaceImplementation): Should use VerifyItemIsAbsentAsync instead after test infrastructure is changed to actually build two distinct projects from the XML 
+            await VerifyItemExistsAsync(markup, "Goo2()");
+            await VerifyItemExistsAsync(markup, "Prop2");
+        }
```
Comment (jcouv) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProviderTests.cs]: >} [](start = 8, length = 1)

It looks like we also don't complete keywords (like `private`) in interfaces. Could be a follow-up PR.

File: src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProvider.cs [hunk 2]
```diff
@@ -121,23 +121,23 @@ private bool IsPreviousTokenValid(SyntaxToken tokenBeforeType)
...
+                       IsClassOrStructOrInterface(memberDeclaration.Parent);
             }
 
             return false;
         }
 
-        private static bool IsClassOrStruct(SyntaxNode node)
-            => node.Kind() == SyntaxKind.ClassDeclaration || node.Kind() == SyntaxKind.StructDeclaration;
+        private static bool IsClassOrStructOrInterface(SyntaxNode node)
+            => node.Kind() == SyntaxKind.ClassDeclaration || node.Kind() == SyntaxKind.StructDeclaration || node.Kind() == SyntaxKind.InterfaceDeclaration;
```
Comment (jcouv) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProvider.cs]: >node.Kind() [](start = 15, length = 11)

nit: I think you could do `=> node.IsKind(SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration, SyntaxKind.InterfaceDeclaration);`

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-ide Please review

**Comment by AlekseyTs** [comment]
@jasonmalinowski, @dotnet/roslyn-ide Please review

**Comment by AlekseyTs** [comment]
@jasonmalinowski, @dotnet/roslyn-ide Please review

**Pull request merged**